phac-nml/irida-next
# Metadata Summary Update by Transfers and Deletions

## What does this PR do and why?
Note: Dependent and branched off of [PR 339](https://github.com/phac-nml/irida-next/pull/339)

This PR is the secondary part of the `metadata_summary` update and includes updating based on group/project/sample `transfers` and `deletion`.

When transferred or deleted, all namespaces  will be updated accordingly.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
In the Rails console, set up your metadata as described in [PR 339](https://github.com/phac-nml/irida-next/pull/339)

Using the website UI, transfer groups, projects, and samples and subsequently check in the console that both old and new parents/ancestors are updated accordingly. Ie: if a sample with `metadata = {'metadatafield1' => 'value1'}` is transferred, ensure its old parents have `metadata_summary['metadatafield1'] -= 1` and the new parents have `metadata_summary['metadatafield1'] += 1` 

Using the website UI, delete groups, projects, and samples and subsequently check in the console that itsparents/ancestors are updated accordingly. Ie: if a sample with `metadata = {'metadatafield1' => 'value1'}` is deleted, ensure its parents have `metadata_summary['metadatafield1'] -= 1` 

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
